src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -823,14 +666,14 @@ public SolutionState WithRunAnalyzers(ProjectId projectId, bool runAnalyzers)
         /// Create a new solution instance with the project specified updated to include
         /// the specified project references.
         /// </summary>
-        public SolutionState AddProjectReferences(ProjectId projectId, IReadOnlyCollection<ProjectReference> projectReferences)
+        public (SolutionState, ProjectState) AddProjectReferences(ProjectId projectId, IReadOnlyCollection<ProjectReference> projectReferences)
         {
+            var oldProject = GetRequiredProjectState(projectId);
```

Comment:
note, the callsite validates this before calling into here.  so it's safe to do so, even in the case of an empty project-references collection.  we changed teh logic slightly here since we need to return the project state with the call as well.